runner.go

```diff
@@ -0,0 +1,90 @@
+package main
+
+import (
+	"errors"
+	"fmt"
+	"log"
+	"os"
+	"os/signal"
+	"syscall"
+	"time"
+)
+
+type runner struct {
+	configuration      *configuration
+	backend            backend
+	cancelChan         chan bool
```

Comment:
Would it make sense to use a context here?
That might be easier to understand.
If we are not using a context, we should close the channel and ensure only one routine closes it. Otherwise, a bad read somewhere will break the cancelling

Reply:
cancelChan is reserved for testing purposes. This should be documented. Revised.